src/server/storage.rs

```diff
@@ -74,7 +74,7 @@ impl ModelStorage {
         let key = model_key(account_id, lattice_id, model.name());
         trace!(%key, "Storing manifest at key");
         let data = serde_json::to_vec(&model).map_err(anyhow::Error::from)?;
-        if let Some(revision) = current_revision {
+        if let Some(revision) = current_revision.filter(|r| r > &0) {
```

Comment:
I don't think I understand this. Is the initial revision for a model always 0?

Reply:
Effectively here, having a revision of `0` and having a `None` revision should have the same outcome. So the initial revision is `0`, and we don't want to use `update` there we want to use `put` to create the model in the first place. This was just fixing an issue where initial create would have `None`, but after deleting a model the revision would come back from the entry API as `Some(0)` and would fail the update.